tests/e2e-kubernetes/testsuites/credentials.go

```diff
@@ -424,32 +457,49 @@ func (t *s3CSICredentialsTestSuite) DefineTests(driver storageframework.TestDriv
 					return assignPolicyToServiceAccount(ctx, sa, policyARN)
 				}
 
-				createPodWithServiceAccountAndPolicy := func(ctx context.Context, policyARN string, allowDelete bool) (*v1.Pod, *v1.ServiceAccount) {
+				createPodWithServiceAccountAndPolicy := func(ctx context.Context, policyARN string, allowDelete bool, asNonRoot bool) (*v1.Pod, *v1.ServiceAccount) {
 					By("Creating Pod with ServiceAccount")
 
 					mountOptions := []string{fmt.Sprintf("region %s", DefaultRegion)}
 					if allowDelete {
 						mountOptions = append(mountOptions, "allow-delete")
 					}
+					if asNonRoot {
+						mountOptions = append(mountOptions,
+							"allow-other",
+							fmt.Sprintf("uid=%d", defaultNonRootUser),
+							fmt.Sprintf("gid=%d", defaultNonRootGroup))
+					}
 					vol := createVolumeResourceWithMountOptions(enablePodLevelIdentity(ctx), l.config, pattern, mountOptions)
 					deferCleanup(vol.CleanupResource)
 
 					sa := createServiceAccountWithPolicy(ctx, policyARN)
 
-					pod, err := createPodWithServiceAccount(ctx, f.ClientSet, f.Namespace.Name, []*v1.PersistentVolumeClaim{vol.Pvc}, sa.Name)
-					framework.ExpectNoError(err)
-					deferCleanup(func(ctx context.Context) error { return e2epod.DeletePodWithWait(ctx, f.ClientSet, pod) })
+					var podModifiers []func(*v1.Pod)
+					podModifiers = append(podModifiers, func(pod *v1.Pod) {
+						pod.Spec.ServiceAccountName = sa.Name
+					})
+					if asNonRoot {
+						podModifiers = append(podModifiers, podModifierNonRoot)
+					}
+
+					pod := createPod(ctx, vol, podModifiers...)
 
 					return pod, sa
 				}
 
 				It("should use pod's service account's read-only role", func(ctx context.Context) {
-					pod, _ := createPodWithServiceAccountAndPolicy(ctx, iamPolicyS3ReadOnlyAccess, false)
+					pod, _ := createPodWithServiceAccountAndPolicy(ctx, iamPolicyS3ReadOnlyAccess, false, false)
```

Comment:
Is there a way to pass in keyword arguments here? Right now I can't understand what these boolean arguments are for.

Reply:
Go doesn't support named arguments, usually structs are used if such a thing is desired, but I think it'd make this more verbose. Since this is a local helper function and not used outside this test-case, I think I'd prefer current one.